api/tests/integration/edge_api/identities/test_edge_identity_featurestates_viewset.py

```diff
@@ -11,28 +11,88 @@
 from tests.integration.helpers import create_mv_option_with_api
 
 
-def test_edge_identities_feature_states_list(
+def test_edge_identities_feature_states_list_does_not_call_sync_identity_document_if_not_needed(
     admin_client,
     environment,
     environment_api_key,
     identity_document,
     dynamo_wrapper_mock,
+    mocker,
 ):
     # Given
     dynamo_wrapper_mock.get_item_from_uuid_or_404.return_value = identity_document
+    sync_identity_document = mocker.patch(
+        "edge_api.identities.views.sync_identity_document"
+    )
+    identity_uuid = identity_document["identity_uuid"]
+    url = reverse(
+        "api-v1:environments:edge-identity-featurestates-list",
+        args=[environment_api_key, identity_uuid],
+    )
+    # When
+    response = admin_client.get(url)
+    # Then
+    dynamo_wrapper_mock.get_item_from_uuid_or_404.assert_called_with(identity_uuid)
+    assert response.status_code == status.HTTP_200_OK
+    assert len(response.json()) == 3
+    assert len(sync_identity_document.mock_calls) == 0
```

Comment:
should this not be `sync_identity_document.delay.assert_not_called()` ?

Reply:
done